src/reporters/search/csvSearchReporter.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import os from 'node:os';
+import fs from 'node:fs';
+import { SearchResult } from '@jsforce/jsforce-node';
+import { SearchReporter } from './reporter.js';
+
+export class CsvSearchReporter extends SearchReporter {
+  public constructor(props: SearchResult) {
+    super(props);
+  }
+
+  public display(): void {
+    if (this.types.length === 0) {
+      this.ux.log('No Records Found');
+    }
+    this.types.map((type) => {
+      const filtered = this.result.searchRecords.filter((t) => t.attributes?.type === type);
+      // remove 'attributes' property from result and csv output
+      filtered.map((r) => delete r.attributes);
```

Comment:
both the filter and the removal of attributes seem common to both classes.  Probably a shared fn?

Or, if you do move to the Map idea, then the value could be filtered to remove attributes via kit/omit `.map((r) => omit(r, 'attributes'))`;